xskillscore/core/probabilistic.py

```diff
@@ -820,3 +822,245 @@ def _reliability(o, f, bin_edges):
+                varlist = list(forecasts.data_vars)
+                if len(varlist) == 1:
+                    v = varlist[0]
+                else:
+                    raise ValueError(
+                        f"Only works for `xr.Dataset` with one variable, found {forecasts.data_vars}. Considering looping over `data_vars` or `.to_array()`."
+                    )
+                f_bin = forecasts[v]
+            else:
+                f_bin = forecasts
+            f_bin = f_bin.stack(ndim=forecasts.dims)
+            f_bin = f_bin.sortby(-f_bin)
+            bin_edges = np.append(f_bin[0] + 1, f_bin)
+            bin_edges = np.unique(bin_edges)  # ensure that in ascending order
+        else:
+            raise ValueError("If bin_edges is str, it can only be continuous.")
+    else:
+        bin_edges = np.sort(bin_edges)  # ensure that in ascending order
+
+    # loop over each bin_edge and get true positive rate and false positive rate from contingency
+    tpr, fpr = [], []
+    for i in bin_edges:
+        dichotomous_category_edges = np.array(
+            [-np.inf, i, np.inf]
+        )  # "dichotomous" means two-category
+        dichotomous_contingency = Contingency(
+            observations,
+            forecasts,
+            dichotomous_category_edges,
+            dichotomous_category_edges,
+            dim=dim,
+        )
+        fpr.append(dichotomous_contingency.false_alarm_rate())
+        tpr.append(dichotomous_contingency.hit_rate())
+    tpr = xr.concat(tpr, "probability_bin")
+    fpr = xr.concat(fpr, "probability_bin")
+    tpr["probability_bin"] = bin_edges
+    fpr["probability_bin"] = bin_edges
+
+    fpr = fpr.fillna(1.0)
+    tpr = tpr.fillna(0.0)
+
+    # pad (0,0) and (1,1)
+    fpr_pad = xr.concat(
+        [
+            xr.ones_like(fpr.isel(probability_bin=0, drop=False)),
+            fpr,
+            xr.zeros_like(fpr.isel(probability_bin=-1, drop=False)),
+        ],
+        "probability_bin",
+    )
+    tpr_pad = xr.concat(
+        [
+            xr.ones_like(tpr.isel(probability_bin=0, drop=False)),
+            tpr,
+            xr.zeros_like(tpr.isel(probability_bin=-1, drop=False)),
+        ],
+        "probability_bin",
+    )
+
+    # https://github.com/scikit-learn/scikit-learn/blob/42aff4e2edd8e8887478f6ff1628f27de97be6a3/sklearn/metrics/_ranking.py#L916
+    # Attempt to drop thresholds corresponding to points in between and
+    # collinear with other points. These are always suboptimal and do not
+    # appear on a plotted ROC curve (and thus do not affect the AUC).
+    # Here xr.diff(_, 2) is used as a "second derivative" to tell if there
+    # is a corner at the point. (...) This keeps all cases where the point should be
+    # kept, but does not drop more complicated cases like fps = [1, 3, 7],
+    # tps = [1, 2, 4]; there is no harm in keeping too many thresholds.
+    if drop_intermediate and fpr.probability_bin.size > 2:
+
+        def _drop_intermediate(fpr, tpr):
```

Comment:
It's kind of awkward to have a function nested in a function within a logic statement plus 8 spaces.

Reply:
So rather exclude the function to utils or to prob.py or contingency.py? Unsure whether another function will need this function. 
or is the long doc comment above the problem?